src/components/Categories/CategoriesTable.tsx

```diff
@@ -0,0 +1,133 @@
+import DeleteCategoryDialog from "./DeleteCategoryDialog";
+import CategoryFormDialog from "./CategoryFormDialog";
+import { CategoriesTableProps } from "../../types";
+import TableSortLabel from "@material-ui/core/TableSortLabel";
+import useSortTable from "../../customHooks/useSortTable";
+
+const styles = (theme: Theme) =>
+  createStyles({
+    root: {
+      width: "95%",
+    },
+    container: {
+      maxHeight: 280,
+    },
+    tableCell: {
+      minWidth: 170,
+      align: "left",
+      padding: theme.spacing(0.2, 0.5, 0.2, 1.5),
+    },
+  });
+
+const CategoriesTable: React.FC<CategoriesTableProps> = ({
+  classes,
+  columns = [],
+  rows: incomingRows,
+  onDelete: handleDelete,
+  onEdit: handleEdit,
+}) => {
+  const rows = useMemo(() => incomingRows.slice(), [incomingRows]);
+  const [page, setPage] = React.useState(0);
+  const [rowsPerPage, setRowsPerPage] = React.useState(4);
+  const { createSortHandler, stableSort, order, orderBy } = useSortTable(
+    columns[0].id
+  );
+
+  const handleChangePage = (event: unknown, newPage: number) => {
+    setPage(newPage);
+  };
+
+  const handleChangeRowsPerPage = (
+    event: React.ChangeEvent<HTMLInputElement>
+  ) => {
+    setRowsPerPage(+event.target.value);
+    setPage(0);
+  };
+
+  return (
+    <Paper className={classes.root}>
+      <TableContainer className={classes.container}>
+        <Table stickyHeader aria-label="sticky table">
+          <TableHead>
+            <TableRow>
+              {columns.map((column) => (
+                <TableCell
+                  key={column.id}
+                  align={column.align}
+                  style={{ minWidth: column.minWidth }}
+                >
+                  <TableSortLabel
+                    active={orderBy === column.id}
+                    direction={orderBy === column.id ? order : "asc"}
+                    onClick={createSortHandler(column.id)}
+                  >
+                    {column.label}
+                  </TableSortLabel>
+                </TableCell>
+              ))}
+              <TableCell className={classes.tableCell}>{"Actions"}</TableCell>
+            </TableRow>
+          </TableHead>
+          <TableBody>
+            {stableSort(rows, order, orderBy)
+              .slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
+              .map((row) => {
+                return (
+                  <TableRow hover role="checkbox" tabIndex={-1} key={row.id}>
+                    {columns.map((column) => {
+                      const value = row[column.id];
+                      return (
+                        <TableCell
+                          align={column.align}
+                          className={classes.tableCell}
+                        >
+                          {value}
+                        </TableCell>
+                      );
+                    })}
+                    <TableCell className={classes.tableCell}>
```

Comment:
if we didnt have this column, then we could use this Table component to almost render any table with any number of columns. and currently we can use it to render any table with delete/edit actions only 👍